model/TestCenterService.php

```diff
@@ -201,4 +204,58 @@ public function getTestCenter($testCenterUri)
     {
         return new core_kernel_classes_Resource($testCenterUri);
     }
+
+    /**
+     * @param core_kernel_classes_Resource $testCenter
+     * @param User $user
+     * @param core_kernel_classes_Resource $role
+     * @return bool
+     * @throws TestCenterException
+     */
+    public function assignUser(\core_kernel_classes_Resource $testCenter, User $user, \core_kernel_classes_Resource $role)
+    {
+        $userResource = $this->getResource($user->getIdentifier());
+        $userRoles = $userResource->getPropertyValues($this->getProperty(UserRdf::PROPERTY_ROLES));
+
+        if (!in_array($role->getUri(), $userRoles)) {
+            throw new TestCenterException(__('User with given role cannot be assigned to the test center.'));
+        }
+
+        switch ($role->getUri()) {
```

Comment:
Switch could be simplified, if you introduce an URI map, something similar like this:
```
    const ASSIGN_URI_MAP = [
        self::ROLE_TESTCENTER_ADMINISTRATOR => ProctorManagementService::PROPERTY_ADMINISTRATOR_URI,
        ProctorService::ROLE_PROCTOR => ProctorManagementService::PROPERTY_ASSIGNED_PROCTOR_URI,
    ];
```

And then you can do:
```
if (!isset(self::ASSIGN_URI_MAP[$role->getUri()])) {
    throw new TestCenterException(__('Role is not allowed.'));
}

return $userResource->editPropertyValues(
    $this->getProperty(self::ASSIGN_URI_MAP[$role->getUri()]),
    $testCenter
); 
```

What do you think?

Reply:
There will be one more PR in taoSync extension to be able to assign sync managers to TC. Test managers require a bit more logic to be assigned (some operations with organnization id) and  there will be more login in those `case`s than only getting the property.

Reply:
Okay, makes sense.